src/d3a/models/area/event_dispatcher.py

```diff
@@ -108,13 +109,14 @@ def _should_dispatch_to_strategies(self, event_type):
             return self.area.events.is_connected and self.area.events.is_enabled
 
     def event_listener(self, event_type: Union[MarketEvent, AreaEvent], **kwargs):
+        current_tick = kwargs.pop('current_tick', None)
```

Comment:
Same as above, I do not think you really need this.

Reply:
I did it else i will get exception of unexpected keyword argument
`TypeError: event_activate() got an unexpected keyword argument 'current_tick'`

Reply:
The, IMO, the `area.activate()` method should be made uniform and have the optional keyword "current_tick" , instead of poping it here.

Reply:
`current_tick` is an optional argument in `area.activate` https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/area/__init__.py#L258

The issue is that if I won't pop it here, then I need to comply all respective callback methods to accept arbitrary keyword arguments like `def cycle_markets(self, .., **kwargs), def activate(self, **kwargs),  def event_tick(self, **kwargs), def event_activate(self, **kwargs)`
Let me know if this is preferable?

Reply:
Sorry, but you only need it in the `event_activate` and `activate` methods, right? If so, I would vote for adding this.

Reply:
I would be using **kwargs for `event_activate ` which itself also don't need it so IMO all callback should be homogenous in terms of either all accepting `**kwargs` or none

Reply:
Hmm, I dont agree here. As we handle all events differently, `event_activate` and `activate` could contain kwargs and the others don't. But if you like, add it to all of them.
Also tagging @spyrostz here to maybe add what is best practice here. Thanks!

Reply:
I agree with Hannes here. First of all, you do not need to pass kwargs on every event, you only need to do so to the event_activate. Second, I think that the correct solution would be to pass it as a normal argument. Have a look at other events like evet_offer, we constantly pass different arguments to different events. This is going to produce a lot of changes, however IMO it is the correct solution, because you really want to pass something via the event here.